pyroute2/netlink/rtnl/tcmsg/cls_fw.py

```diff
@@ -38,5 +43,16 @@ class options(nla, nla_plus_police):
                ('TCA_FW_CLASSID', 'uint32'),
                ('TCA_FW_POLICE', 'police'),  # TODO string?
                ('TCA_FW_INDEV', 'hex'),  # TODO string
-               ('TCA_FW_ACT', 'hex'),  # TODO
+               ('TCA_FW_ACT', 'tca_act_prio'),
                ('TCA_FW_MASK', 'uint32'))
+
+    class tca_act_prio(nla):
```

Comment:
We should try to not duplicate this code everywhere for each filter. We could perhaps introduce a common ancestor class with this kind of code?

Reply:
After looking at the code and made some experiments with code factorization, it seems to be hard to do what you are looking for.

Look at the function 'compile_nla' in 'netlink/__init__.py' file: sadly, it looks up the class by its name using 'getattr(**self**, "tca_act_prio")' so it will never find the declaration of the 'tca_act_prio' outside of the current class.

Moreover, it might be due to my knowledge limitations of Python but I don't know if it will be easy to override sub-classes declared inside 'tca_act_prio' class. For reference, I invite you to compare 'action' declarations of sched_bpf.py, act_fw.py and cls_u32.py.
Each declaration of the 'tca_act_prio' class is different from the others.

We can talk about it tomorrow.